os/mm/umm_heap/umm_free.c

```diff
@@ -82,10 +82,12 @@
 
 void free(FAR void *mem)
 {
-	int heap_idx;
-	heap_idx = mm_get_heapindex(mem);
-	if (heap_idx != INVALID_HEAP_IDX) {
-		mm_free(&USR_HEAP[heap_idx], mem);
+	struct mm_heap_s *heap;
+	heap = mm_get_heap(mem);
+	if (heap) {
+		return mm_free(heap, mem);
```

Comment:
this function's return type is void, so you'd better change like below:
```
if (heap) {
    mm_free(xx,x);
    return;
}
```

Reply:
fixed.